src/main/java/org/apache/mesos/state/CuratorStateStore.java

```diff
@@ -150,48 +139,54 @@ public void storeTasks(Collection<Protos.TaskInfo> tasks) throws StateStoreExcep
     }
 
     @Override
-    public Collection<Protos.TaskInfo> fetchTasks(String execName) throws StateStoreException {
-        Collection<Protos.TaskInfo> taskInfos = new ArrayList<>();
+    public void storeStatus(Protos.TaskStatus status) throws StateStoreException {
```

Comment:
Spoke to you in person.  We must not store TaskStatus's with irrelevant TaskIDs.